addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -0,0 +1,352 @@
+        """
+        Convert the invoice lines to a list of items required for the Facturae xml generation
+        :return:                        A tuple containing the Face items, the taxes and the invoice totals data.
+        """
+        self.ensure_one()
+        items = []
+        totals = {
+            'total_gross_amount': 0.,
+            'total_general_discounts': 0.,
+            'total_general_surcharges': 0.,
+            'total_taxes_withheld': 0.,
+            'total_tax_outputs': 0.,
+            'total_payments_on_account': 0.,
+            'amounts_withheld': 0.,
+        }
+        taxes = []
+        taxes_withheld = []
+        for line in self.invoice_line_ids:
+            invoice_line_values = {}
+
+            discount = max(0., (line.discount * line.price_unit / 100))
+            surcharge = abs(min(0., (line.discount * line.price_unit / 100)))
+            total_price = line.price_subtotal + discount - surcharge
+            gross_amount = line.price_subtotal
+            totals['total_gross_amount'] += gross_amount
+            totals['total_general_discounts'] += discount
+            totals['total_general_surcharges'] += surcharge
+            base_line = self.env['account.tax']._convert_to_tax_base_line_dict(
+                line, partner=line.partner_id, currency=line.currency_id, product=line.product_id, taxes=line.tax_ids,
+                price_unit=line.price_unit, quantity=line.quantity, discount=line.discount, account=line.account_id,
+                price_subtotal=line.price_subtotal, is_refund=line.is_refund, rate=line.amount_currency/line.balance if line.balance else 0.
+            )
+
+            taxes_computed = self.env['account.tax']._compute_taxes([base_line])
+            taxes_withheld_computed = [tax for tax in taxes_computed["tax_lines_to_add"] if tax["tax_amount"] < 0]
+            taxes_normal_computed = [tax for tax in taxes_computed["tax_lines_to_add"] if tax["tax_amount"] >= 0]
+
+            taxes_output = [self._convert_computed_tax_to_template(tax) for tax in taxes_normal_computed]
+            totals['total_tax_outputs'] += sum((abs(tax["tax_amount"]) for tax in taxes_normal_computed))
+
+            tax_withheld_output = [self._convert_computed_tax_to_template(tax) for tax in taxes_withheld_computed]
+            totals['total_taxes_withheld'] += sum((abs(tax["tax_amount"]) for tax in taxes_withheld_computed))
+
+            invoice_line_values.update({
+                'ItemDescription': line.name,
+                'Quantity': line.quantity,
+                'UnitOfMeasure': line.product_uom_id.l10n_es_edi_facturae_uom_code,
+                'UnitPriceWithoutTax': total_price / line.quantity,
+                'TotalCost': total_price,
+                'DiscountsAndRebates': [{
+                    'DiscountReason': '',
+                    'DiscountRate': f'{line.discount:.2f}',
+                    'DiscountAmount': discount
+                }, ] if discount != 0. else [],
```

Comment:
we could just have a variable discount being negative or positive

Reply:
Already taken care of